File: backend/src/main/java/com/healthcare/dashboard/entities/Investment.java
package com.healthcare.dashboard.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "investments")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Investment {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private String nom;
    
    @Column(nullable = false)
    private String categorie; // EQUIPEMENT, INFRASTRUCTURE, TECHNOLOGIE, FORMATION
    
    @Column(columnDefinition = "TEXT")
    private String description;
    
    @Column(nullable = false)
    private Double montant;
    
    @Column(nullable = false)
    private LocalDateTime dateInvestissement;
    
    private LocalDateTime dateFinPrevue;
    
    @Column(nullable = false)
    private String statut; // PLANIFIE, EN_COURS, TERMINE, ANNULE
    
    private String fournisseur;
    
    private String responsable;
    
    @Column(columnDefinition = "TEXT")
    private String beneficesAttendus;
    
    private Double retourInvestissement; // ROI en pourcentage
    
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: flutter_app/lib/models/investment.dart
class Investment {
  final int? id;
  final String nom;
  final String categorie;
  final String description;
  final double montant;
  final DateTime dateInvestissement;
  final DateTime? dateFinPrevue;
  final String statut;
  final String? fournisseur;
  final String? responsable;
  final String? beneficesAttendus;
  final double? retourInvestissement;
  final String? niveauRisque;
  final double? roiEstime;
  final DateTime? dateEcheance;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  Investment({
    this.id,
    required this.nom,
    required this.categorie,
    required this.description,
    required this.montant,
    required this.dateInvestissement,
    this.dateFinPrevue,
    required this.statut,
    this.fournisseur,
    this.responsable,
    this.beneficesAttendus,
    this.retourInvestissement,
    this.niveauRisque,
    this.roiEstime,
    this.dateEcheance,
    this.createdAt,
    this.updatedAt,
  });

  factory Investment.fromJson(Map<String, dynamic> json) {
    return Investment(
      id: json['id'],
      nom: json['nom'] ?? '',
      categorie: json['categorie'] ?? '',
      description: json['description'] ?? '',
      montant: (json['montant'] ?? 0).toDouble(),
      dateInvestissement: DateTime.parse(json['dateInvestissement']),
      dateFinPrevue: json['dateFinPrevue'] != null ? DateTime.parse(json['dateFinPrevue']) : null,
      statut: json['statut'] ?? 'PLANIFIE',
      fournisseur: json['fournisseur'],
      responsable: json['responsable'],
      beneficesAttendus: json['beneficesAttendus'],
      retourInvestissement: json['retourInvestissement']?.toDouble(),
      niveauRisque: json['niveauRisque'],
      roiEstime: json['roiEstime']?.toDouble(),
      dateEcheance: json['dateEcheance'] != null ? DateTime.parse(json['dateEcheance']) : null,
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
      updatedAt: json['updatedAt'] != null ? DateTime.parse(json['updatedAt']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'nom': nom,
      'categorie': categorie,
      'description': description,
      'montant': montant,
      'dateInvestissement': dateInvestissement.toIso8601String().split('T')[0],
      if (dateFinPrevue != null) 'dateFinPrevue': dateFinPrevue!.toIso8601String().split('T')[0],
      'statut': statut,
      if (fournisseur != null) 'fournisseur': fournisseur,
      if (responsable != null) 'responsable': responsable,
      if (beneficesAttendus != null) 'beneficesAttendus': beneficesAttendus,
      if (retourInvestissement != null) 'retourInvestissement': retourInvestissement,
      if (niveauRisque != null) 'niveauRisque': niveauRisque,
      if (roiEstime != null) 'roiEstime': roiEstime,
      if (dateEcheance != null) 'dateEcheance': dateEcheance!.toIso8601String().split('T')[0],
    };
  }
}
